src/Doctrine/Common/State/PersistProcessor.php

```diff
@@ -52,10 +52,34 @@ public function process(mixed $data, Operation $operation, array $uriVariables =
         // https://github.com/doctrine/orm/issues/8461#issuecomment-1250233555
         if ($operation instanceof HttpOperation && HttpOperation::METHOD_PUT === $operation->getMethod() && ($operation->getExtraProperties()['standard_put'] ?? false)) {
             \assert(method_exists($manager, 'getReference'));
-            // TODO: the call to getReference is most likely to fail with complex identifiers
             $newData = $data;
             if (isset($context['previous_data'])) {
-                $newData = 1 === \count($uriVariables) ? $manager->getReference($class, current($uriVariables)) : clone $context['previous_data'];
+                // the correct entity need to be fetched, we need to get the ORM ids from previous data
+                $reflectionProperties = $this->getReflectionProperties($context['previous_data']);
+                $ormIds = [];
+                \assert(method_exists($manager, 'getClassMetadata'));
+                foreach ($manager->getClassMetadata($class)->getIdentifier() as $identifier) {
+                    if (\array_key_exists($identifier, $reflectionProperties)) {
+                        $value = $reflectionProperties[$identifier]->getValue($context['previous_data']);
+                    } else {
+                        // This id is not from current class, let's move up in class hierarchy
```

Comment:
This is a behavior I want to remove from API Platform and that is done at https://github.com/api-platform/core/blob/9b4b58ca0113a2645b58a116fe9e4bf200df8aa3/src/Api/IdentifiersExtractor.php#L96. 

See this function is quite wrong already as it is "guessing" that identifiers are at the correct place. The correct way do to this is that in your `resourceClass` you have a property marked as `ApiProperty(identifier: true)`. It can be a ghost property that is just fetching `parent::getId()` but it should be stringable.

Example:

```php
class Foo { protected $id; }

class Bar extends Foo {
  #[ApiProperty(identifier: true)]
  private string $myCustomId;
  
  // must be accessible/readable!
  public function getMyCustomId(): string {
      return parent::$id;
  }
}
```

Reply:
Ok, i think i get your point here, i'll try to change this so it uses IdentifierExtractor instead. 

`ApiProperty(identifiers: true)`  doesn't seem to apply to the current problem though : the id doctrine needs is not the one used in the URI in my case.